graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/autoconfig/ExcludeAutoConfigurationsEnvironmentPostProcessor.kt

```diff
@@ -59,8 +59,15 @@ class ExcludeAutoConfigurationsEnvironmentPostProcessor : EnvironmentPostProcess
             )
     }
 
-    private fun extractAllExcludes(propertySources: MutablePropertySources): String =
-        propertySources
+    private fun extractAllExcludes(propertySources: MutablePropertySources): String {
+        if (propertySources.any { it.name == INLINED_TEST_PROPERTIES }) {
+            val testExclude = propertySources.find { it.name == INLINED_TEST_PROPERTIES }?.getProperty(EXCLUDE)
+            if (testExclude is String && testExclude.isNotBlank()) {
+                return testExclude
+            }
+        }
```

Comment:
to avoid double-iteration over `propertySources`:
```suggestion
    private fun extractAllExcludes(propertySources: MutablePropertySources): String {
        val testExclude = propertySources.find { it.name == INLINED_TEST_PROPERTIES }?.getProperty(EXCLUDE)
        if (testExclude != null && testExclude is String && testExclude.isNotBlank()) {
            return testExclude
        }
```